front/src/containers/SearchPage/SearchPage.tsx

```diff
@@ -723,17 +723,21 @@ class SearchPage extends React.Component<SearchPageProps, SearchPageState> {
     return response;
   };
   updateSearchParams = async (params) => {
-    this.setState({	
-      ...this.state,	
-      params: { ...(this.state?.params || {}), ...params },	
+    this.setState({
+      ...this.state,
+      params: { ...(this.state?.params || {}), ...params },
     });
     const variables = { ...this.state.params, ...params };
     const { data } = await this.props.mutate({ variables });
     const searchQueryString = new URLSearchParams(
       this.props.history.location.search
     );
     const siteViewUrl = searchQueryString.getAll('sv').toString() || 'default';
-    const pageViewUrl = searchQueryString.getAll('pv').toString() || 'default';
+    // This assumes that the site provider is not passing a url into the page
+    // view fragment portion of the query otherwise we would need to call the
+    //  page view query without passing the url into it to retrieve the default url
```

Comment:
@rodctech Can you go into more detail on this? The code makes sense, I'm just not sure if I understand the comment.

Reply:
@hifilorau i think you'd have a better detailed understanding on this for the comment

Reply:
@kaveer-k @rodctech in theory you could get a non-default pageview returned from the site provider query if you were to pass a pageview url into the siteprovider query at the pageview level. We don't as of now, and so it always returns the default. i just wanted to document here that the pageview query could return a non-default pageview when there isn't a url passed in. We don't do that now and i'd think would always use the pageview query itself to do this but we could so wanted to make a note in the event we ever ran into an issue.

Reply:
@hifilorau Ah okay, I think I get it. I probably need to understand the `SiteProvider` component a bit more but can you give me an example use case where we would want to override the default using the `url` argument? Could it be used to provide different defaults for different sub-sites?

Also, I don't think we've officially met so nice to meet you :)

Reply:
@kaveer-k likewise, nice to meet you as well! I think originally when it was written we thought we might also end up using this query to get different pageviewss or siteviews as needed so we gave it the ability to pass a url in but this siteprovider query has SO much data in it and is really a performance killer, that I think we all agree that a siteview query and a pageview query are the way to go and we can strip alot of this out of the siteprovider query in time. 

I think though if you were using a sub-site it would have a different site id so that it would automatically give you its default based on that. Honestly, I can't think of any use cases for the siteprovider where we would want to pass urls into either of those (not that i couldn't be wrong) so we should probably actually take the url out for both pageviews and siteviews when we have independent queries set up.

Reply:
@hifilorau Not sure I follow this. Let's discuss on call tomorrow.

Reply:
I gave this a re-read and I think I can clarify for @kaveer-k and it may be helpful for the whole team to confirm what we are trying to accomplish and some history.

History: every site has a "default" site_view - this should be stored with the flag in the site_view model where one site_view per site has a default value and this could be updated via a checkbox on the admin site_view overview or configuration page - however, for some reason, the site_view "default" is hard coded to HAVE to have a value of "default" in the parameter value that is displayed in the url (as sv=default). Because of this behavior, I think the line to set the sv=default, if it wasn't otherwise validly specified in the querystring, was put in place to prevent errors in the lookup of the site_view. (note, the more proper name should have been "search_views" but there was a site_views table/model that was co-opted for this purpose - we haven't changed because of the vision of page_views described below)

Issue #404 outlines the change to not require "default" as the name of the default site_view - which would simplify configuration and copying of site_views.

When we created page_views, we did set that up to use the default checkbox and each page has a default, regardless of what the name of that default value that will appear in the querystring as pv=study1 for example. However, we were having issues with errors when no value was supplied in the querystring. The "solution" for that was to add a similar line of code as for sv that inserted default when missing. An alternative might have been (should have been) to just tolerate missing values for pv (or sv) and look up the default value if it is needed and not supplied rather than updating the url querystring with a hard coded "default" as the value, thus locking us into the already awkward convention that the default sv has to be named default and perpetuating that to the pv.

Finally, we absolutely want/need the ability to change the value of the sv or pv via the url querystring, because this is how we build the links for buttons to move between card or list or table views and between presearch views, etc. We also will increasingly have links from the study page back to searches where we want to construct, in mailmerge, links that selectively use part of the querystring and replace other parts. 

As to the comment in the code, I think the value of sv or pv is EITHER supplied in the querystring or it is the default value. I can't think of where it would be "looked up" via the siteprovider or pageprovider other than finding the default when nothing was specified. ie the order should be "does querystring have a valid sv or pv, if not, use the default one".

Note, the vision and idea for "page_views" is that the site can be self building and each "type" of page can be configured from the same interface, so we may eventually merge "site_views" into "page_views" as the "search" type instead of "study" type - the type would specify the route ie /search and /study and specifying the root graphql search and key - this would/could allow building intervention page, institution page, profile and admin and dashboard pages. (the "default" would need to be adjusted to be one unique default for each site/type)